resources/assets/js/videos/videoContainer.vue

```diff
@@ -425,7 +425,9 @@ export default {
             this.attachAnnotationLabel(annotation)
                 .then(() => {
                     if (lastLabel) {
-                        this.detachAnnotationLabel(annotation, lastLabel);
+                        this.detachAnnotationLabel(annotation, lastLabel)
+                            .then(Events.$emit('video.swap', annotation));
```

Comment:
I would prefer to do it the same than in the image annotation tool here. Add a "hard" method call to `this.$refs.videoScreen.refreshAnnotation()` to the `detachAnnotationLabel()` method and be done with it. While I've implemented this global `Event` bus at some point, I wouldn't use it any more because you never know from where the event originated and where the listeners are.